core/src/main/scala/com/salesforce/op/stages/impl/feature/DateToUnitCircleTransformer.scala

```diff
@@ -30,22 +30,39 @@
 
 package com.salesforce.op.stages.impl.feature
 import com.salesforce.op.features.types._
-import com.salesforce.op.stages.base.unary.UnaryTransformer
+import com.salesforce.op.stages.base.sequence.{SequenceEstimator, SequenceModel, SequenceTransformer}
+import com.salesforce.op.utils.spark.{OpVectorColumnMetadata, OpVectorMetadata}
 import com.salesforce.op.{FeatureHistory, UID}
-import com.salesforce.op.utils.spark.OpVectorMetadata
 import enumeratum.{Enum, EnumEntry}
-
-import scala.reflect.runtime.universe.TypeTag
 import org.apache.spark.ml.linalg.Vectors
-import org.apache.spark.ml.param.Param
+import org.apache.spark.ml.param.{Param, Params}
+import org.apache.spark.sql.Dataset
 import org.joda.time.{DateTime => JDateTime}
 
+import scala.reflect.runtime.universe.TypeTag
+
+trait DateToUnitCircleParams extends Params {
+
+  final val timePeriod: Param[String] = new Param[String](parent = this,
+    name = "timePeriods",
+    doc = "The time period to extract from the timestamp",
+    isValid = (value: String) => TimePeriod.values.map(_.entryName).contains(value)
+  )
+
+  setDefault(timePeriod, TimePeriod.HourOfDay.entryName)
+
+  /** @group setParam */
+  final def getTimePeriod: TimePeriod = TimePeriod.withName($(timePeriod))
+
+  final def setTimePeriod(value: TimePeriod): this.type = set(timePeriod, value.entryName)
+}
+
 /**
  * Following: http://webspace.ship.edu/pgmarr/geo441/lectures/lec%2016%20-%20directional%20statistics.pdf
  * Transforms a Date or DateTime field into a cartesian coordinate representation
  * of an extracted time period on the unit circle
  *
- * @param timePeriod The time period to extract from the timestamp
+ * parameter timePeriod The time period to extract from the timestamp
```

Comment:
do you mean spark parameter?

Reply:
yes - it bugged me that it was highlighted red